src/Swift.Bindings/src/Model/TypeSpecAttribute.cs

```diff
@@ -0,0 +1,37 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+using System;
+using System.Collections.Generic;
+using System.Text;
+using System.Linq;
+
+namespace BindingsGeneration;
+
+public class TypeSpecAttribute
+{
+    public TypeSpecAttribute(string name)
+    {
+        Name = name;
+        Parameters = new List<string>();
```

Comment:
Could you provide some examples here? Is the string a good type here?

Reply:
It isn't, but it's not the worst. In this case, the contents of attributes are incredibly varied. In the spec, it's described as a Dyck language (balanced brackets, braces, etc) but not a lot of other detail. It looks like some of the common attributes are preparsed with the information that we care about set aside.  Mostly we care about the attributes for a couple really important things: `@available` and `@escaping`. The latter has no parameters, while the former has very predictable ones.